### ISSUE TITLE
Improve JSON/JX/JC encoding consistency for unbacked buffers

### ISSUE THREAD

**Issue by svaarala** [svaarala/duktape#869]
Currently the behavior for Node.js Buffers and other buffer types is different in JSON/JX/JC modes.

**Comment by fatcerberus** [comment]
Not sure what you mean by unbacked buffers?

**Comment by svaarala** [comment]
You can create e.g. a 20-byte Uint8Array over a dynamic plain buffer, and the resize that plain buffer to, say, 4 bytes. The conceptual 20-byte index space is no longer "backed" by the underlying plain buffer. This never happens for standards compliant code but the situation can happen via the C API.

**Comment by fatcerberus** [comment]
Ah, I hadn't considered dynamic plain buffers.  That makes sense.  Good to know that situation was anticipated and it's memory safe.

**Comment by svaarala** [comment]
Yeah, it's somewhat a "user error" situation in some sense, so the only guarantees I've tried to give is that it's memory safe. Other than that behavior varies quite a bit; but when something is changed (like buffer JSON behavior in this case) it's good to try to improve the consistency of the behavior.

**Comment by fatcerberus** [comment]
The most reasonable behavior to me would be to serialize the unbacked part as though it is filled with 0 bytes.
